### ISSUE TITLE
[jest/valid-title] false negatives then passing a string variable instead of inlining string directly

### ISSUE THREAD

**Issue by JustFly1984** [jest-community/eslint-plugin-jest#878]
mini example:

```ts
import { getEstimateBudgetAndTotal } from './shared'

// eslint-disable-next-line jest/valid-title
describe(getEstimateBudgetAndTotal.name, () => {
  it('returns zero if passed no data', () => {
    expect(getEstimateBudgetAndTotal()).toMatchInlineSnapshot(`
      Object {
        "totalBudget": 0,
        "totalEstimate": 0,
        "totalPaid": 0,
      }
    `)
  })
})
```

**Comment by G-Rath** [comment]
Would you please provide more information about what you're reporting, including the behaviour you're expecting and your eslint config?

**Comment by G-Rath** [comment]
@nitzanashi correct, but we can't know that `getEstimateBudgetAndTotal` is a function as that requires type information.

```
export const getEstimateBudgetAndTotal = {
  name: {
    firstName: ...,
    lastName: ...
  }
}
```

The above will have the exact same AST in your code snippet in ESLint.

To allow this you can either switch to TypeScript (since that can do what this rule does much better) or enable the `ignoreTypeOfDescribeName` option on the rule.

**Comment by JustFly1984** [comment]
Please reopen, the project is actually in typescript. The issue is with your plugin, it should check the type.

**Comment by G-Rath** [comment]
I've thought about using type information to improve this rule, but realised this boils down to wanting to enforce either:
  1. consistent use of literal strings only as titles (aka disallowing passing in non-literal strings or literal numbers)
  2. ensure that the correct type is always being passed as titles (which is `number | string | Function | FunctionLike`)

The first is exactly what a linting rule is good for and what this rule can do; The second is exactly what TypeScript is for and do a lot better than a linting rule.

Now yes we could use type information to extend this rule to allow it to enforce 2. but that would require TypeScript to be available  meaning you might as well just rely on TypeScript instead to handle this as it's going to be much better at enforcing this.

So I think the solution here is that you should be using `ignoreTypeOfDescribeName` to have this rule not try and check the type at all.

**Comment by JustFly1984** [comment]
@G-Rath Thanx!
